### ISSUE TITLE
Feature request: support forcing ipv4/ipv6

### ISSUE THREAD

**Issue by Seirdy** [ducaale/xh#274]
`cURL` supports `-4`/`--ipv4` and `-6`/`--ipv6` flags. Quoting the `curl(1)` manpage:

``` text
       -4, --ipv4
              This option tells curl to resolve names to IPv4 addresses only, and
              not for example try IPv6.

              Example:
               curl --ipv4 https://example.com

              See also --http1.1 and --http2. This option overrides -6, --ipv6.

       -6, --ipv6
              This option tells curl to resolve names to IPv6 addresses only, and
              not for example try IPv4.

              Example:
               curl --ipv6 https://example.com

              See also --http1.1 and --http2. This option overrides -4, --ipv4.
```

For testing purposes, having this feature would be really convenient.

**Comment by ducaale** [comment]
I haven't verified it yet but It seems the HTTP library we are using (i.e reqwest) supports forcing either IPv4 or IPv6 by setting the client's `local_address`, see https://github.com/seanmonstar/reqwest/issues/584#issuecomment-780916982.

@Seirdy would you be interested in opening a PR for this feature?

**Comment by blyxxyz** [comment]
HTTPie issue: httpie/httpie#94

**Comment by zuisong** [comment]
> I haven't verified it yet but It seems the HTTP library we are using (i.e reqwest) supports forcing either IPv4 or IPv6 by setting the client's `local_address`, see [seanmonstar/reqwest#584 (comment)](https://github.com/seanmonstar/reqwest/issues/584#issuecomment-780916982).
> 
> @Seirdy would you be interested in opening a PR for this feature?

I would like to try it. 😄

**Comment by ducaale** [comment]
Resolved in #276
